### PR TITLE
RoR Capstone

### PR DESCRIPTION
**Pull request by lucascomamala**
- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
  - Should be built with Devise.
- Food list (only for a group that has less than 3 members):
  - Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
  - Should lead to a form that allows users to add new food.
- Recipes list:
  - Should display a list of recipes created by the logged-in user as in the wireframe.
  - Should lead to recipe details.
  - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
  - Should display a list of all public recipes ordered by newest as in the wireframe.
  - Should lead to recipe details.
- Recipe details:
  - Should display a toggle button that allows for a recipe to be made public or private.
  - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
  - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
  - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
  - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- Write unit and integration tests

Capture
Capture
Capture
Capture
Capture
Capture
Capture

### CODE REVIEW

File: app/views/recipes/_recipe.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="card text-bg-dark mb-4" id="<%= dom_id recipe %>">
+  <div class="card-body">
+    <h3 class="card-title"><%= recipe.name %></h3>
```
Comment (Lizdev-05) [inline app/views/recipes/_recipe.html.erb]: - [x] Please make sure that clicking on a recipe item takes you to the recipe details page `/recipes/:id/` as shown in the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/recipe_food.png?raw=true) :pray:

File: Gemfile [hunk 2]
```diff
@@ -50,12 +50,18 @@ gem "bootsnap", require: false
...
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
-  gem "debug", platforms: %i[ mri mingw x64_mingw ]
+  gem 'debug', platforms: %i[mri mingw x64_mingw]
+  gem 'factory_bot_rails'
+  gem 'faker'
+  gem 'rails-controller-testing'
+  gem 'rspec-rails'
+  gem 'rubocop', '>= 1.0', '< 2.0'
 end
```
Comment (Lizdev-05) [inline Gemfile:59]: - [x] I suggest that you use the Capybara testing library for the testing of your views :+1: 
  
  ```ruby
   #Gemfile
  
   group :development, :test do
     ...
     gem 'capybara'
     gem 'rspec-rails'
   end
  ```

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @lucascomamala , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 2 above)
Comment (lucascomamala) [reply Gemfile:59]: I'm using it! Line 75 in the Gemfile :)

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @lucascomamala 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] [OPTIONAL] This is a group project, but I haven't seen any commits from your coding partner. Please ensure to always work together during group projects, however if your coding partner is not available and you have to work solo, kindly provide proof from student success 👍 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**